types/tinymce/index.d.ts

```diff
@@ -633,7 +633,7 @@ export interface Formatter {
 
   matchNode(node: html.Node, name: string, vars: {}, similar: boolean): {};
 
-  register(name: {}, format?: {}): void;
+  register(name: {} | string, format?: {} | object[]): void;
```

Comment:
Additionally `{} | object[]` is equivalent to `{}`, and `object | object[]` would be equivalent to `object` (since `object[]` also satisfies `object`).

Some tests would help here.